src/gen-ts.ts

```diff
@@ -72,21 +79,39 @@ export async function generateDeclarations(
 function analyzerToAst(
     analysis: analyzer.Analysis, config: Config, rootDir: string):
     ts.Document[] {
-  const exclude = (config.exclude || ['test/**', 'demo/**'])
-                      .map((p) => new minimatch.Minimatch(p));
+  const exclude =
+      (config.exclude || defaultExclude).map((p) => new minimatch.Minimatch(p));
   const addReferences = config.addReferences || {};
   const removeReferencesResolved = new Set(
       (config.removeReferences || []).map((r) => path.resolve(rootDir, r)));
   const renameTypes = new Map(Object.entries(config.renameTypes || {}));
 
+  // For consistency, we want to produce a `d.ts` file for every HTML file, even
+  // when it has no JavaScript and hence produces no typings. When an HTML file
+  // does have JavaScript, Analyzer will produce two separate documents: a
+  // `js-document` and an `html-document`, both of which contain the JavaScript
+  // features. To avoid producing double typings, only consider
+  // `html-documents` for which there was no corresponding `js-document`.
+  const analyzerDocs: Set<analyzer.Document> =
```

Comment:
Hm, I think we should instead filter by toplevel documents, as there's guaranteed to be at most one toplevel document at a given URL, but there could be multiple inline html or js documents at any given URL

Reply:
I think I already account for the fact that there could be multiple JS docs with the same URL in the loop below -- and also multiple files with the same d.ts name (e.g. foo.html and foo.js are combined to produce foo.d.ts).

Reply:
Hm, fair, so maybe everything is fine. Do we have any tests of the tricky cases, like multiple inline scripts in the same file? Or html inside of js inside of html, or js inside of html inside of js?

Reply:
Added "combo" test to show that `combo.js` and multiple script tags in `combo.html` are correctly combined into a single `combo.d.ts`. From offline discussion simplified (and fixed) the logic so that we just ignore inline documents.

PTAL